lib/ansible/module_utils/network/nxos/config/telemetry/telemetry.py

```diff
@@ -0,0 +1,287 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2019 Cisco and/or its affiliates.
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+The nxos_telemetry class
+It is in this file where the current configuration (as dict)
+is compared to the provided configuration (as dict) and the command set
+necessary to bring the current configuration to it's desired end-state is
+created
+"""
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import re
+from copy import deepcopy
+
+from ansible.module_utils.network.common.cfg.base import ConfigBase
+from ansible.module_utils.network.common.utils import to_list
+from ansible.module_utils.network.nxos.facts.facts import Facts
+from ansible.module_utils.network.nxos.cmdref.telemetry.telemetry import TMS_GLOBAL, TMS_DESTGROUP, TMS_SENSORGROUP, TMS_SUBSCRIPTION
+from ansible.module_utils.network.nxos.utils.telemetry.telemetry import normalize_data, remove_duplicate_context
+from ansible.module_utils.network.nxos.utils.telemetry.telemetry import valiate_input, get_setval_path
+from ansible.module_utils.network.nxos.utils.telemetry.telemetry import get_module_params_subsection
+from ansible.module_utils.network.nxos.nxos import NxosCmdRef, normalize_interface
+
+
+class Telemetry(ConfigBase):
+    """
+    The nxos_telemetry class
+    """
+
+    gather_subset = [
+        '!all',
+        '!min',
+    ]
+
+    gather_network_resources = [
+        'telemetry',
+    ]
+
+    def __init__(self, module):
+        super(Telemetry, self).__init__(module)
+
+    def get_telemetry_facts(self):
+        """ Get the 'facts' (the current configuration)
+
+        :rtype: A dictionary
+        :returns: The current configuration as a dictionary
+        """
+        facts, _warnings = Facts(self._module).get_facts(self.gather_subset, self.gather_network_resources)
+        telemetry_facts = facts['ansible_network_resources'].get('telemetry')
+        if not telemetry_facts:
+            return {}
+        return telemetry_facts
+
+    def edit_config(self, commands):
+        return self._connection.edit_config(commands)
+
+    def execute_module(self):
+        """ Execute the module
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        result = {'changed': False}
+        commands = list()
+        warnings = list()
+        module_params = self._module.params['config']
+
+        # Normalize interface name.
+        int = module_params.get('source_interface')
+        if int:
+            module_params['source_interface'] = normalize_interface(int)
+
+        existing_telemetry_facts = self.get_telemetry_facts()
+        commands.extend(self.set_config(existing_telemetry_facts))
+        if commands:
+            if not self._module.check_mode:
+                self.edit_config(commands)
+                # self._connection.load_config(commands)
```

Comment:
How do we know when to use `edit_config` versus `load_config`? It looks like `load_config` actually calls `edit_config`.

Reply:
In this case this is boiler plate code.  I have load_config commented out because once we add nxapi and httpapi we will need to call `load_config` since `edit_config` is currently only available for network_cli if I remember correctly.

Reply:
ah, okay. Can you add that comment to the code? Got a feeling we'll forget the why if it's not commented.